mentat/code_feature.py

```diff
@@ -268,36 +236,27 @@ def get_code_message_from_features(features: list[CodeFeature]) -> list[str]:
         if len(path_features) == 1:
             code_message += path_features[0].get_code_message()
         else:
-            code_message += get_code_message_from_intervals(path_features)
+            code_message += _get_code_message_from_intervals(path_features)
     return code_message
 
 
 def get_consolidated_feature_refs(features: list[CodeFeature]) -> list[str]:
-    """Return a list of 'path:<interval>,<interval>' strings"""
+    """
+    Return a list of 'path:<interval>,<interval>' strings, merging code features with the same path
+    """
     level_info_by_path = defaultdict[Path, list[Interval | None]](list)
     for f in features:
-        if f.level == CodeMessageLevel.CODE:
+        if f.interval.whole_file():
             level_info_by_path[f.path].append(None)
-        elif f.level == CodeMessageLevel.INTERVAL:
-            level_info_by_path[f.path].append(f.interval)
         else:
-            pass  # Skipping filename, code_maps
+            level_info_by_path[f.path].append(f.interval)
 
     consolidated_refs = list[str]()
     for path, level_info in level_info_by_path.items():
         ref_string = str(path)
-        if not any(level is None for level in level_info):
-            intervals = sorted(
-                [level for level in level_info if isinstance(level, Interval)],
-                key=lambda i: i.start,
-            )
-            ref_string += f":{intervals[0].start}-"
-            last_end = intervals[0].end
-            for i in intervals[1:]:
-                if i.start > last_end + 1:
-                    ref_string += f"{last_end},{i.start}-"
-                last_end = i.end
-            ref_string += str(last_end)
+        intervals = sorted([level for level in level_info if level is not None])
+        if intervals and None not in level_info:
+            ref_string += ":" + ",".join(str(interval) for interval in intervals)
```

Comment:
I changed how this works; before, we used to merge separate intervals together (so 1-10 and 10-20 would become 1-20), but I disabled this because I think it makes sense to show the user the actual intervals (otherwise they might try to exclude 1-20, but that isn't actually included so they wouldn't be able to). I think this makes sense but let me know what y'all think.